src/Symfony/Component/DependencyInjection/Compiler/ResolveNamedArgumentsPass.php

```diff
@@ -64,6 +65,28 @@ protected function processValue($value, $isRoot = false)
                 throw new InvalidArgumentException(sprintf('Unable to resolve service "%s": method "%s()" has no argument named "%s". Check your service definition.', $this->currentId, $class !== $this->currentId ? $class.'::'.$method : $method, $key));
             }
 
+            $lastResolvedArgument = \end($resolvedArguments);
+
+            if (\is_array($lastResolvedArgument) &&
+                !empty($lastResolvedArgument) &&
+                method_exists('ReflectionParameter', 'isVariadic')
+            ) {
+                try {
+                    $reflection = $this->getReflectionMethod($value, $method);
```

Comment:
I think a better implementation would hook into the existing foreach loop on line 56-57.
We just need a `if (\PHP_VERSION_ID >= 50600 && $p->isVariadic()` check inside the loop.
No need for enforcing this is the "last argument", PHP already did so.
And the key of the added params must be explicitly set (`$resolvedArguments[$j++]` instead of `$resolvedArguments[]`, which can be wrong)